src/stream_base.cc

```diff
@@ -364,76 +264,30 @@ int StreamBase::WriteString(const FunctionCallbackInfo<Value>& args) {
 
   buf = uv_buf_init(data, data_size);
 
-  if (!IsIPCPipe()) {
-    err = DoWrite(req_wrap, &buf, 1, nullptr);
-  } else {
-    uv_handle_t* send_handle = nullptr;
-
-    if (!send_handle_obj.IsEmpty()) {
-      HandleWrap* wrap;
-      ASSIGN_OR_RETURN_UNWRAP(&wrap, send_handle_obj, UV_EINVAL);
-      send_handle = wrap->GetHandle();
-      // Reference LibuvStreamWrap instance to prevent it from being garbage
-      // collected before `AfterWrite` is called.
-      CHECK_EQ(false, req_wrap->persistent().IsEmpty());
-      req_wrap_obj->Set(env->handle_string(), send_handle_obj);
-    }
-
-    err = DoWrite(
-        req_wrap,
-        &buf,
-        1,
-        reinterpret_cast<uv_stream_t*>(send_handle));
+  uv_stream_t* send_handle = nullptr;
+
+  if (IsIPCPipe() && !send_handle_obj.IsEmpty()) {
+    // TODO(addaleax): This relies on the fact that HandleWrap comes first
+    // as a superclass of each individual subclass.
+    // There are similar assumptions in other places in the code base.
+    // A better idea would be having all BaseObject's internal pointers
+    // refer to the BaseObject* itself; this would require refactoring
+    // throughout the code base but makes Node rely much less on C++ quirks.
```

Comment:
@TimothyGu Since you were asking for ideas, this might be a decent medium-sized project?